standards/SPIFFE-ID.md

```diff
@@ -135,7 +141,47 @@ In order for an SVID to be considered valid, it MUST leverage a document type fo
+
+Example: if trust domains A and B agree to the meaning of the "environment" attribute, that agreement of semantics does not extend to trust domain C which may also happen to include an attribute with the same name of "environment".
+
+#### 4.1.4. Veracity
+SVID consumers residing in the same trust domain as the SVID being consumed can generally assume that assertions in the SVID are true, as the consumer likely considers the authority that issued the SVID to be wholly trusted. It does not necessarily mean however, that assertions of similar form and nature can be assumed true when originating from a different trust domain.
+
+Operators should carefully consider, on a case by case basis, whether or not a given foreign authority should be considered trustworthy in the context of a specific assertion.
+
+For example, imagine that the operator of trust domain A and trust domain B have a shared customer and a business relationship. During normal operation in trust domain A, a specialized process authenticates the customer and the trust domain’s authority creates an SVID that embeds an assertion indicating the customer’s identity. The purpose of this process is to mitigate unauthorized access to customer data in the event that an intermediary service within the trust domain is compromised by proving that the customer was authenticated and did in fact make this request.
+
+Now imagine that a request for the customer’s data is received by the storage service in trust domain A, except the caller presented an SVID from trust domain B. Even though the presented SVID may have the necessary assertion indicating that the shared customer was authenticated and authorized the request, it is a bad idea to blindly assume that trust domain B has indeed authenticated your customer. If trust domain B is compromised, it could claim to have authenticated any user it wishes, thus creating the very circumstance that the measure was designed to mitigate in the first place.
+
+Therefore, any assertion appearing in an SVID that is not explicitly defined by the SPIFFE specification set should be considered generally untrustworthy when the assertion is made by a foreign authority, unless it can be rigorously proven otherwise (e.g. if trust domain A and B have shared infrastructure, or if trust domain A is permitted to audit trust domain B).
```

Comment:
I don't think "that is not explicitly defined by the SPIFFE specification set" is the right qualification here.

The question administrators need to ask themselves is just a blanket, "what assertions do I trust this domain to make?"  They should NOT just assume that SPIFFE-defined assertions are true!  The difference between SPIFFE-defined and not is in how an operator should react.  If the operator does not trust SPIFFE-defined assertions, then maybe they should not federate at all.  If they trust SPIFFE-defined assertions, but not some other assertions, then it might be fine to federate and just filter out those other assertions.

Reply:
I am worried that we are trending down the highly skeptical security person who always says no and never believes anyone path. I updated this paragraph with the latest commit to exclude the quote you called out, but now the text basically says "don't trust anything in an svid unless you otherwise know it to be likely true". I think this puts our users in a very tough spot, and gives the impression that they should not trust this technology because everything in it is effectively hearsay. It feels too extreme.

Reply:
I think this goes back to the other comment thread on just being clear about business agreements vs security.
You should never trust anything you don't have to. But if your business A decides to federate with B, now you have to trust them. Your business relationship defines somewhere what elements of each other's SVIDs you will rely on and for what purposes, these elements being things you are otherwise **blindly trusting unless signed by an unknown/untrusted CA, or notified of compromise**, because that is what your business agreement is.

One doesn't even have to go into federation for this concept. For A's own processes to talk to other A processes, the receiver is also just going off of whatever internal agreements there are for usage.

Similar to @evan2645 's concern, IMO in trying to clarify this situation we have made usage seem less and less reliable. For instance, in the example for "a B process says it has authenticated A's customer": in operational usage or permissioning models I wouldn't even consider this something to bring up, because in whatever authorization model I have as a security employee of A I only allow `spiffe://a.ca/customer/authenticator` to state it has authenticated A's customers. And no matter how hard it might try to be otherwise, `spiffe://**b**.ca/customer/authenticator` is not the same identity.

Reply:
I'm actually happy just dropping this sentence entirely.  In the same section we have this sentence

> Operators should carefully consider, on a case by case basis, whether or not a given foreign authority should be considered trustworthy in the context of a specific assertion.

Which I think conveys the sensible advice we want to give, without the highly skeptical language: "considered generally untrustworthy unless it can be rigorously proven otherwise."

Reply:
> I think this puts our users in a very tough spot, and gives the impression that they should not trust this technology because everything in it is effectively hearsay. It feels too extreme.

I agree with @evan2645's concern. I think we want to point out to people that they should scrutinize assertions for trustworthiness without conveying tinfoil hat levels of paranoia. You have to be able to trust _something_.

Accordingly, I would also be happy to drop this sentence.

Reply:
Ok great, I think that works well in resolving my concern. It has been stricken